include/usb/class/usb_audio.h

```diff
@@ -0,0 +1,286 @@
+};
+
+struct usb_audio_feature_unit_evt {
+	struct device *dev;
+	enum usb_audio_direction dir;
+	enum usb_audio_feature_unit_control_selectors cs;
+	u8_t channel;
+	void *val;
+};
+
+/**
+ * @brief Callback type used to inform the app that data were requested
+ *	  from the device and may be send to the Host.
+ *	  For sending the data usb_audio_send() API function should be used.
+ *
+ * @note User may not use this callback and may try to send in 1ms task
+ *	 instead. Sending every 1ms may be unsuccessful and may return -EAGAIN
+ *	 if Host did not required data.
+ *
+ * @param dev The device for which data were requested by the Host.
+ */
+typedef void (*usb_audio_data_request_cb_t)(const struct device *dev);
+
+/**
+ * @brief Callback type used to inform the app that data were successfully
+ *	  send/received.
+ *
+ * @param dev	 The device for which the callback was called.
+ * @param buffer Pointer to the net_buf data chunk that was successfully
+ *		 send/received.
+ * @param size	 Amount of data that were successfully send/received.
+ */
+typedef void (*usb_audio_data_completion_cb_t)(const struct device *dev,
+					       struct net_buf *buffer,
+					       size_t size);
+
+/**
+ * @brief Callback type used to inform the app that Host has changed
+ *	  one of the features configured for the device.
+ *	  Applicable for all devices.
+ *
+ * @warning Host may not use all of configured features.
+ *
+ * @param evt Event information to be parsed by app.
+ */
+typedef void (*usb_audio_feature_updated_cb_t)(
+					struct usb_audio_feature_unit_evt evt);
+
+/**
+ * @brief Audio callbacks used to interact with audio devices by user App.
+ *
+ * usb_audio_ops structure contains all relevant callbacks to interact with
+ * USB audio devices. Each of this callbacks is optional and may be left NULL.
+ * This will not break the stack but could make USB audio device useless.
+ * Depending on the device some of those callbacks are necessary to make USB
+ * device work as expected sending/receiving data. For more information refer
+ * to callback documentation above.
+ */
+struct usb_audio_ops {
+	/* Callback called when data could be send */
+	usb_audio_data_request_cb_t data_request_cb;
+
+	/* Callback called when data were successfully written with sending
+	 * capable device. Applicable for headset and microphone. Unused for
+	 * headphones.
+	 */
+	usb_audio_data_completion_cb_t data_written_cb;
+
+	/* Callback called when data were successfully received by receive
+	 * capable device. Applicable for headset and headphones. Unused for
+	 * microphone.
+	 */
+	usb_audio_data_completion_cb_t data_received_cb;
+
+	/* Callback called when features were manipulation by Host */
+	usb_audio_feature_updated_cb_t feature_update_cb;
+};
+
+/** USB Audio device controls
+ *
+ * This structure present the structure of possible features that USB audio
+ * device may support. Enabling features is done via DT overlay.
+ */
+struct usb_audio_features {
```

Comment:
Why is this structure defined in the public API? Is the application supposed to somehow access it?

Reply:
Hmm, no. I will put this to the internals.